sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -1172,6 +1172,7 @@ func (p *provider) Invoke(tok tokens.ModuleMember, args resource.PropertyMap) (r
 		Label:         fmt.Sprintf("%s.args", label),
 		KeepSecrets:   p.acceptSecrets,
 		KeepResources: p.acceptResources,
+		KeepUnknowns:  true,
```

Comment:
Talking out loud here, does this mean that all provider invokes must be written in a way that understands unknowns? Is that currently the case? Should this be configurable by provider? Might be good to try this patch out with some invokes non-terraform based providers like kubernetes and azure-native.

Reply:
Thinking specifically about the behavior in previews.

Reply:
Thank you - this is what I'm trying to figure out 🙂  Should provider impl be shielded from unknowns completely, is that something we're trying to maintain? If I had confirmation here it would be easier to make decisions on the concrete code things.

In the concrete layer...  this patch is now a bit outdated as I can't trigger this path anymore. What happens is this chain:

```
pulumi-user-program --> resmon.Invoke --> provider.Invoke
```

resmon.Invoke currently does KeepUnknowns: true but provider.Invoke does KeepUnknowns: false. It's inconsistent but it only would matter if these functions actually get passed unknowns.

There was a bug in how KeepUnkonwn: false behaved that I fixed
https://github.com/pulumi/pulumi/pull/7214/files

(The function should have been better named "filterOutUnknowns").

So I think right now it works by provider.Invoke filtering out unknowns and not resmon.Invoke filtering out unknowns. If this is the right division of responsibilities we can keep it as is.

The other option mentioned in this PR was RejectUnknowns: true in resmon.Invoke. This would fail with an exception instead of filtering them out.